src/UI/Art/SpriteArtist/UnitSpriteSet.java

```diff
@@ -284,7 +285,9 @@ public void drawUnitIcons(Graphics g, GameInstance game, Unit u, int animIndex,
     }
 
     // Transport icon.
-    if( u.heldUnits != null && !u.heldUnits.isEmpty() )
+    boolean fogOn = map.game.isFogEnabled();
+    boolean hideCargo = fogOn && u.CO.isEnemy(map.viewer);
+    if( !hideCargo && !u.heldUnits.isEmpty() )
```

Comment:
The carts replace the envelope icon with an always-active dash icon, which I'm not a super fan of.
I'm not a huuuuge fan of just hiding the icon, but bleh.

Also, adding an icon now would be annoying with #417 pending